modules/account-lib/src/coin/stx/transaction.ts

```diff
@@ -0,0 +1,171 @@
+  deserializeTransaction,
+  addressToString,
+  StacksMessageType,
+  createStacksPublicKey,
+  SingleSigSpendingCondition,
+  isSingleSig,
+  StacksPublicKey,
+  TransactionAuthField
+} from '@stacks/transactions';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { SigningError } from '../baseCoin/errors';
+import { BaseKey } from '../baseCoin/iface';
+import { BaseTransaction, TransactionType } from '../baseCoin';
+import { SignatureData, TxData } from './iface';
+import { getTxSenderAddress, bufferToHexPrefixString, removeHexPrefix } from './utils';
+import { KeyPair } from './keyPair';
+
+export class Transaction extends BaseTransaction {
+  private _stxTransaction: StacksTransaction;
+  protected _type: TransactionType;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  canSign(key: BaseKey): boolean {
+    // TODO: implement this validation method
+    return true;
+  }
+
+  async sign(keyPair: KeyPair[] | KeyPair): Promise<void> {
+    const keyPairs = keyPair instanceof Array ? keyPair : [keyPair];
+    const signer = new TransactionSigner(this._stxTransaction);
+    for (const kp of keyPairs) {
+      const keys = kp.getKeys(true);
+      if (!keys.prv) {
+        throw new SigningError('Missing private key');
+      }
+      const privKey = createStacksPrivateKey(keys.prv);
+      signer.signOrigin(privKey);
+    }
+  }
+
+  async appendOrigin(keyPair: KeyPair): Promise<void> {
+    const keys = keyPair.getKeys(true);
+    if (!keys.pub) {
+      throw new SigningError('Missing public key');
+    }
+    const pubKey = createStacksPublicKey(keys.pub);
+    const signer = new TransactionSigner(this._stxTransaction);
+    signer.appendOrigin(pubKey);
+  }
+
+  async signWithSignatures(signature: SignatureData): Promise<void> {
+    if (!signature) {
+      throw new SigningError('Missing signatures');
+    }
+    this._stxTransaction = this._stxTransaction.createTxWithSignature(signature.data);
+  }
+
+  /**
+  * Get the signatures associated with this transaction.
+  */
+  get signature(): string[] {
+    if (this._stxTransaction && this._stxTransaction.auth.spendingCondition) {
+      if (isSingleSig(this._stxTransaction.auth.spendingCondition)) {
+        return [this._stxTransaction.auth.spendingCondition.signature.data]
+      } else {
+        return this._stxTransaction.auth.spendingCondition.fields.map(this.getSignatureFromField)
+      }
+    }
+    return [];
+  }
+
+  private getSignatureFromField(field: TransactionAuthField): string {
+    switch (field.contents.type) {
+      case StacksMessageType.PublicKey:
+        return field.contents.data.toString('hex')
+      case StacksMessageType.MessageSignature:
+        return field.contents.data
+    }
+  }
+
+  /** @inheritdoc */
+  toJson() {
+    const result: TxData = {
+      id: this._stxTransaction.txid(),
+      fee: new BigNumber(this._stxTransaction.auth.getFee().toString()).toNumber(),
```

Comment:
What's the point of going from string to bigNumber to number? Think we can just getFee here in Numerical value (can't imagine us getting overprecise on the fees)

Reply:
Updated.